libs/payments/ui/src/lib/client/components/Header/index.tsx

```diff
@@ -0,0 +1,372 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import Image from 'next/image';
+import helpIcon from '@fxa/shared/assets/images/help.svg';
+import mozillaIcon from '@fxa/shared/assets/images/moz-logo-bw-rgb.svg';
+import bentoIcon from '@fxa/shared/assets/images/bento.svg';
+import defaultAvatarIcon from '@fxa/shared/assets/images/avatar-default.svg';
+import closeIcon from '@fxa/shared/assets/images/close.svg';
+import desktopIcon from '@fxa/shared/assets/images/desktop.svg';
+import mobileIcon from '@fxa/shared/assets/images/mobile.svg';
+import monitorIcon from '@fxa/shared/assets/images/monitor.svg';
+import relayIcon from '@fxa/shared/assets/images/relay.svg';
+import vpnIcon from '@fxa/shared/assets/images/vpn-logo.svg';
+import pocketIcon from '@fxa/shared/assets/images/pocket.svg';
+import signOutIcon from '@fxa/shared/assets/images/sign-out.svg';
+import { LinkExternal } from '@fxa/shared/react';
+import { Localized, useLocalization } from '@fluent/react';
+import { useState } from 'react';
+import { useClickOutsideEffect } from '../../hooks/useClickOutsideEffect';
+import { useEscKeydownEffect } from '../../hooks/useEscKeydownEffect';
+import { User } from 'next-auth';
+import { constructHrefWithUtm } from '../../../utils/constructHrefWithUtm';
+
+export const Header = ({
+  authContext,
+}: {
+  authContext?: { user: User | undefined; signOut: () => Promise<any> };
+}) => {
+  let localized: boolean;
+  const signedIn = authContext && authContext.user;
+
+  try {
+    // useLocalization provides the simplest check of whether we're within the LocalizationProvider context
+    const l10n = useLocalization();
+    if (!l10n) {
+      throw new Error('Not in the localization context');
+    }
+    localized = true;
+  } catch {
+    localized = false;
+  }
+
+  const [isBentoMenuRevealed, setBentoMenuRevealed] = useState(false);
+  const toggleBentoMenuRevealed = () => {
+    setBentoMenuRevealed(!isBentoMenuRevealed);
+    setAvatarMenuRevealed(false);
+  };
+  const bentoMenuInsideRef =
+    useClickOutsideEffect<HTMLDivElement>(setBentoMenuRevealed);
+
+  const [isAvatarMenuRevealed, setAvatarMenuRevealed] = useState(false);
+  const toggleAvatarMenuRevealed = () => {
+    setAvatarMenuRevealed(!isAvatarMenuRevealed);
+    setBentoMenuRevealed(false);
+  };
+  const avatarMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(
+    setAvatarMenuRevealed
+  );
+
+  useEscKeydownEffect(() => {
+    setBentoMenuRevealed(false);
+    setAvatarMenuRevealed(false);
+  });
+
+  const links = {
+    desktop: constructHrefWithUtm(
+      'https://www.mozilla.org/firefox/new/',
```

Comment:
Let's have the base urls for these links in a constants file, similar to Settings.

Ref:
https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/constants/index.tsx#L26-L40

https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/components/Settings/BentoMenu/index.tsx#L48-L91

Reply:
great idea!